packages/nx/src/tasks-runner/utils.ts

```diff
@@ -64,16 +66,34 @@ export function getDependencyConfigs(
   return dependencyConfigs;
 }
 
-function expandDependencyConfigSyntaxSugar(
-  deps: (TargetDependencyConfig | string)[]
+export function expandDependencyConfigSyntaxSugar(
+  deps: (TargetDependencyConfig | string)[],
+  graph: ProjectGraph
 ): TargetDependencyConfig[] {
```

Comment:
It's not really in scope of this PR but could we change this to only parse the syntax sugar?

```suggestion
export function expandDependencyConfigSyntaxSugar(
  deps: string,
  graph: ProjectGraph
): TargetDependencyConfig {
```